pkg/container-collection/options.go

```diff
@@ -270,11 +271,10 @@ func withPodInformer(nodeName string, fallbackMode bool) ContainerCollectionOpti
 					// first: remove containers that are not running anymore
 					nonrunning := k8sClient.GetNonRunningContainers(c)
 					for _, id := range nonrunning {
-						// container had not been added, no need to remove it
-						if _, ok := containerIDs[id]; !ok {
-							continue
-						}
-
+						// Try to remove the container even if it wasn't
+						// captured by the pod informer. It could happen
+						// that it's captured by a start-container hook but
+						// the stop-container one is having issues.
 						cc.RemoveContainer(id)
```

Comment:
Could it happen that `k8sClient.GetNonRunningContainers` returns a container id that has not been started yet? Maybe the container has been started but the API Server or the cache does not know that it is already started, so we would wrongly remove the container immediately after its startup if the API server is a bit slow to update the `Running->StartedAt` field?

Reply:
I think that could happen. I changed the logic a bit, now the container is only removed if this was previously detected by the pod informer, in this way I think we avoid the situation you described.